File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int c1= 0 , c2=0, c3=nums.length-1;
         while ( c2 <= c3)
         {
            if ( nums[c2]==0)
            {
                swap(nums ,c1,c2);
                c1++;
                c2++;
            }
            else if( nums[c2]==1)
            {
                c2++;
            }
            else{
                 swap(nums ,c2,c3);
                 c3--;
            }
         }
    }
     private  void swap(int[] nums, int i , int j)
     {
        int temp = nums[i];
        nums[i]= nums[j];
        nums[j]= temp;
     }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int c0=0;
        int c1=0;
        int c2=0;
        for ( int i=0;i< n ; i++){
           if ( nums[i] ==0)
           {
             c0++;
           }
           else if ( nums[i]==1)
           {
            c1++;
           }
           else
           c2++;
        
        }
        int index= 0;
        for ( int i=0; i< c0; i++){
            nums[index++]=0;        }
             for ( int i=0; i< c1; i++){
            nums[index++]=1;        }
             for ( int i=0; i< c2; i++){
            nums[index++]=2;        }

    }
};